docs/src/main/asciidoc/amazon-snapstart.adoc

```diff
@@ -0,0 +1,184 @@
+com.fasterxml.jackson.core.io.UTF32Reader
+com.fasterxml.jackson.core.json.ByteSourceJsonBootstrapper
+com.fasterxml.jackson.core.json.JsonReadContext
+com.fasterxml.jackson.core.json.JsonWriteContext
+com.fasterxml.jackson.core.json.UTF8StreamJsonParser
+com.fasterxml.jackson.core.JsonEncoding
+com.fasterxml.jackson.core.JsonGenerationException
+com.fasterxml.jackson.core.JsonLocation
+com.fasterxml.jackson.core.JsonStreamContext
+com.fasterxml.jackson.core.JsonToken
+...
+----
+
+The format is simple: one class per line.
+
+=== Computing the class list
+
+NOTE: That step is particularly not user-friendly. We plan to improve it.
+
+To compute the list of classes, we recommend deploying your function and setting the `JAVA_TOOL_OPTIONS` environment variable to `-verbose:class`.
+Then execute your function and retrieve the log (in CloudWatch).
+You should be able to extract the class names using sed/awk or any text editor.
+
+=== Application class list
+
+By default, Quarkus generates the class list of the classes included in your application (including the classes generated by Quarkus).
+So, you do not have to repeat them in the `quarkus-preload-classes.txt` file.
+
+You can disable this feature using:
+
+[source, properties]
+----
+quarkus.snap-start.generate-application-class-list=false
+----
+
+=== Disable preloading
+
+You can disable the preloading of classes using:
+
+[source, properties]
+----
+quarkus.snap-start.preload-classes=false
+----
+
+=== Skipping class initialization
+
+By default, the classes are preloaded, the classes are also _initialized_, meaning it also resolves the dependent classes.
+You can disable this behavior using:
+
+[source, properties]
+----
+quarkus.snap-start.initialize-classes=false
+----
+
+== Client Priming
+
+Client _priming_ is a technic that allows initializing a client during the snapshotting process, so, it's already fully functional during the application runtime.
+
+There are two ways to achieving _priming_:
+
+1. initialize the client in a `static` block, which, thanks to class preloading will be executed before the snapshot
+2. register a CRaC Resource doing the initialization
+
+(1) can be achieved as follows:
+
+[source, java]
+----
+@ApplicationScoped
+public class HeroRepository {
+    private static final DynamoDbClient client;
+
+    static {
+        client = DynamoDbClient.builder()
+                .region(Region.US_EAST_2)
+                .credentialsProvider(DefaultCredentialsProvider.create())
+                .build();
+        client.describeEndpoints();
+    }
+    // ...
+}
+----
+
+The next section covers (2).
+
+== Resource registration
+
+SnapStart uses CRaC API to allow the application to execute custom code before the snapshotting or during the restoration.
+
+IMPORTANT: While it's the CRaC API, SnapStart is not CRaC and can do _things_ that would not work with others CRaC implementations.
+
+[source, java]
+----
+package me.escoffier.hello;
```

Comment:
I think we should stay away of using your name in examples? I think we use `org.acme` elsewhere.